pkg/cloudfoundry/YamlUtil_test.go

```diff
@@ -0,0 +1,418 @@
+package cloudfoundry
+
+import (
+	"fmt"
+	"github.com/ghodss/yaml"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"regexp"
+	"testing"
+	"time"
+)
+
+type fileInfoMock struct {
+	name string
+	data []byte
+}
+
+func (fInfo fileInfoMock) Name() string       { return "" }
+func (fInfo fileInfoMock) Size() int64        { return int64(0) }
+func (fInfo fileInfoMock) Mode() os.FileMode  { return 0444 }
+func (fInfo fileInfoMock) ModTime() time.Time { return time.Time{} }
+func (fInfo fileInfoMock) IsDir() bool        { return false }
+func (fInfo fileInfoMock) Sys() interface{}   { return nil }
+
+func TestFilesRelated(t *testing.T) {
+
+	writeFileCalled := false
+	traverseCalled := false
+
+	var replacements map[string]interface{}
+
+	oldStat := _stat
+	oldReadFile := _readFile
+	oldWriteFile := _writeFile
+	oldTraverse := _traverse
+
+	defer func() {
+		_stat = oldStat
+		_readFile = oldReadFile
+		_writeFile = oldWriteFile
+		_traverse = oldTraverse
+	}()
+
+	reset := func() {
+
+		writeFileCalled = false
+
+		traverseCalled = false
+
+		replacements = make(map[string]interface{})
+
+		_stat = func(name string) (os.FileInfo, error) {
+			return fileInfoMock{}, nil
+		}
+
+		_readFile = func(name string) ([]byte, error) {
+			if name == "manifest.yml" {
```

Comment:
To save some code duplication I would propose to use the filesMock as it has all this already implemented

Reply:
Will have a look and most likely adjust today. The fileUtils mocking was more the less developed in parallel and I was not aware. In general I guess we both have the same aim: not to use the real file system during the tests - which is a high value.

Reply:
@daniel-kurzynski : as already mentioned above, I'm very open to switch to the file mock approach you outlined above.

I played around now and it seems not that easy to use that file utils mock, basically since there is no interface defined so that we can use something in the free wild and exchange it with a mock for the tests. Would it be fine for you to establish an interface accordingly? 

I saw already the approach of having some file utils as part of a bigger utils bundle, see e.g. [here](https://github.com/SAP/jenkins-library/blob/master/cmd/nexusUpload.go#L24). Would it be fine to create an interface for the file system based operations?

Already earlier I tried to start with something like that. See [here](https://github.com/SAP/jenkins-library/blob/master/pkg/piperutils/FileUtils.go#L16). Maybe we can adjust this (would require some adjustments in the already existing code, of course). Depending on the efforts for having such an interface we can either merge that without and refactor later or switch to the new approach before merging that. What do you think?

Reply:
Several points:
* There is already an interface defined in `piperutils.FileUtils`. Although it doesn't actually include everything like the new `Glob()`.
* However, from what I understand how this is intended to work in Go, the idea is not to define an interface for what you provide, but instead you define an interface of what you *need*. This means you declare a private interface with only the sub-set of methods you need from FileUtils. And during runtime, you satisfy this requirement with an instance of the `piperutils.Files` struct. And during testing, you satisfy it with `mock.FilesMock`.

Reply:
>what I understand how this is intended to work in Go, the idea is not to define an interface for what you provide, but instead you define an interface of what you _need_.

Adjusted to the strategy outlined by @stippi2 . There is now an interface dedicate to what we need in the YamlUtils. This covers now `FileRead` and `FileWrite` and uses the mocks from `pkg/mock/file/fileUtils.go

There is now the additional PR #1706 adding some facilities for checking if a file has been written. Needed by the tests here.